templates/boltzmann/core/prelude.ts

```diff
@@ -47,12 +53,74 @@ if (!process.env.HONEYCOMB_TEAM && process.env.HONEYCOMBIO_TEAM) {
   process.env.HONEYCOMB_TEAM = process.env.HONEYCOMBIO_TEAM
 }
 
-beeline({
-  writeKey: process.env.HONEYCOMB_WRITEKEY,
-  dataset: process.env.HONEYCOMB_DATASET,
-  sampleRate: Number(process.env.HONEYCOMB_SAMPLE_RATE) || 1,
-  serviceName,
-})
+if (!process.env.HONEYCOMB_DATASET && process.env.HONEYCOMBIO_DATASET) {
+  process.env.HONEYCOMB_DATASET = process.env.HONEYCOMBIO_DATASET
+}
+
+let sdk: NodeSDK | null = null
+
+if (
+  process.env.HONEYCOMB_WRITE_KEY &&
+  process.env.HONEYCOMB_DATASET &&
+  process.env.HONEYCOMB_API_HOST
+) {
+  const metadata = new Metadata()
+  metadata.set('x-honeycomb-team', process.env.HONEYCOMB_WRITE_KEY)
+  metadata.set('x-honeycomb-dataset', process.env.HONEYCOMB_DATASET)
+
+  const traceExporter = new CollectorTraceExporter({
+    url: process.env.HONEYCOMB_API_HOST,
+    credentials: credentials.createSsl(),
+    metadata
+  })
+
+  sdk = new NodeSDK({
+    resource: new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: serviceName
+    }),
+    traceExporter,
+    instrumentations: [getNodeAutoInstrumentations()],
+    metricInterval: Math.floor(
+      1 / (Number(process.env.HONEYCOMB_SAMPLE_RATE) || 1)
```

Comment:
The idea here is that `f = 1/T` where `T` is the interval/period. I *believe* I checked the units but I don't remember for sure. It would be worth ensuring that the beeline rate is per-second and that the otel interval is seconds, but we should definitely double-check this.

Reply:
Oh, this is a *sampling* rate, not a metrics output rate. 🤦 

So at a minimum, I need to figure out if there's a way to set the *sampling* rate, and if not figure something out. But is it worth making the interval as well? I'd like to punt on that, since we don't configure that in beelines - do you think that's reasonable?

Reply:
I wouldn't worry about the pumping interval for now! The default is fine for now I'm sure